autopilot/contractor/contractor.go

```diff
@@ -1196,6 +1231,82 @@ func performPostMaintenanceTasks(ctx *mCtx, bus Bus, alerter alerts.Alerter, cc
 	return nil
 }
 
+// performV2ContractMigration migrates v1 contracts to v2 contracts once the v2
+// allow height is reached. That means:
+// - we form a v2 contract with all hosts that we currently have a v1 contract with
+// - we delete the v1 contract afterwards
+// - migrations will take over
+func performV2ContractMigration(ctx *mCtx, bus Bus, cr contractReviser, logger *zap.SugaredLogger) {
```

Comment:
This still misses testing to close https://github.com/SiaFoundation/renterd/issues/1546
I'll add that in a separate PR once we have v2 migrations working as well.